pkg/network/node/networkpolicy.go

```diff
@@ -538,16 +597,30 @@ func (np *networkPolicyPlugin) handleDeletePod(obj interface{}) {
 	pod := obj.(*kapi.Pod)
 	glog.V(5).Infof("Watch %s event for Pod %q", watch.Deleted, getPodFullName(pod))
 
-	_, podExisted := np.pods[pod.UID]
+	oldPod, podExisted := np.pods[pod.UID]
 	if !podExisted {
 		return
 	}
 
 	np.lock.Lock()
 	defer np.lock.Unlock()
 
+	// only cleanup the pod ip from current flows
+	glog.V(5).Infof("Delete IP %s flow for Pod %q", pod.Status.PodIP, getPodFullName(pod))
+
+	otx := np.node.oc.NewTransaction()
+	otx.DeleteFlows("ip, nw_dst=%s", oldPod.Status.PodIP)
```

Comment:
This leaves `np.flows` out of sync, which in an extreme edge case could result in the actual flows getting out of sync. (If a pod was deleted and then a new pod added with the same IP, `syncNamespace` might decide it didn't need to call `AddFlow` because the expected flow was already in `np.flows`.)

But if you make `syncNamespace` not call `DumpFlows` as I suggested, then I think you can just remove this optimization here, and just call `refreshNetworkPolicies` and `syncNamespace` like it does now.

Reply:
yes your suggestion makes much more sense, thx.